### PR TITLE
Deal with the case where a VPP doesn't declare vectors

### PR DESCRIPTION
**Pull request by friedmud**
Small fixup for #7819

It's one option for fixing it.  The alternative is to make all of the functions return pointers instead so they can be NULL in the case where there is nothing there.

Doesn't really matter to me either way... but this seemed the least intrusive.

refs #7809

### CODE REVIEW

File: test/src/vectorpostprocessors/LateDeclarationVectorPostprocessor.C [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    GeneralVectorPostprocessor(parameters),
+    _value(nullptr)
+{
+}
+
+void
+LateDeclarationVectorPostprocessor::initialize()
+{
+  if (_t_step == 1)
+    _value = &declareVector("value");
```
Comment (permcody) [inline test/src/vectorpostprocessors/LateDeclarationVectorPostprocessor.C]: I really haven't looked at your code much but are you telling me that `initialSetup()` is too early for your needs? Declaring during execution doesn't seem like the best approach to me.

Comment (friedmud) [inline test/src/vectorpostprocessors/LateDeclarationVectorPostprocessor.C]: Yep - I don't know what data I will have until after I execute the first step...

**Comment by permcody** [comment]
Well I don't like anything that requires the inspection of `_t_step` or `_t`, but I guess that's your code :wink:. These kinds of object make it hard to do integrity checking. Since this doesn't destroy MOOSE in any nasty way it can go in. Thanks for the test case.

**Review by permcody — APPROVED** [review]

**Comment by permcody** [comment]
I figured it wasn't but it's still setting up new storage during the solve process none-the-less. In some sense MOOSE needs to be more late binding but it also just makes troubleshooting more difficult in general.

**Comment by friedmud** [comment]
It definitely does.  I could see how a flag might be useful
"allow_late_binding" or something for VPPs that turns off error checking
for an individual VPP.

**Pull request merged**